packages/compass-utils/src/filesystem.ts

```diff
@@ -0,0 +1,145 @@
+import { promises as fs, createWriteStream } from 'fs';
+import path from 'path';
+import { glob } from 'glob';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { getAppPath } from './electron';
+
+const { log, mongoLogId } = createLoggerAndTelemetry('COMPASS-UTILS');
+
+type FilesystemOptions<T = unknown> = {
+  subdir: string;
+  onSerialize: (content: T) => string;
+  onDeserialize: (content: string) => T;
+};
+
+export class Filesystem<T> {
+  private readonly subdir: string;
+  private readonly onSerialize: FilesystemOptions<T>['onSerialize'];
+  private readonly onDeserialize: FilesystemOptions<T>['onDeserialize'];
+
+  constructor({
+    subdir = '/',
```

Comment:
You probably don't want to default to something that can be accidentally resolved as root folder:

```
path.resolve(__dirname, '/') // returns '/'
```

If default we want is no subdirectory, should this be just empty string?